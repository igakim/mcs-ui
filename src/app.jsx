import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import App from './view/Home';

export default (store) => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('root'),
  );
};
