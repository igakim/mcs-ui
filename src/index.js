import '@babel/polyfill';

// Import Components:
import Button from './components/button';

export { Button };
