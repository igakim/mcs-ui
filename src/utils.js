export const isEqualTypes = (type, compareType) => (
  type && type === compareType
);

export const getButtonType = type => (
  type && `mcs-button-${type}`
);

export const getClassNameFromArray = classesList => classesList
  .filter(el => (el !== undefined && el !== null))
  .join('-');
