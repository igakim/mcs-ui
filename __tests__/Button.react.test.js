import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../src';

test('component should match snapshot', () => {
  const component = renderer.create(
    <Button>Button</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('component should have "mcs-button-primary" class', () => {
  const component = renderer.create(
    <Button type="primary">Button</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('component should be disabled and have "mcs-button-primary" class', () => {
  const component = renderer.create(
    <Button type="primary" disabled>Button</Button>,
  );
  const component2 = renderer.create(
    <Button type="primary" disabled="disabled">Button</Button>,
  );
  const component3 = renderer.create(
    <Button type="primary" disabled={true}>Button</Button>,
  );
  const tree = component.toJSON();
  const tree2 = component2.toJSON();
  const tree3 = component3.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree2).toMatchSnapshot();
  expect(tree3).toMatchSnapshot();
});

test('component should have user classes as well', () => {
  const component = renderer.create(
    <Button type="primary" className="super-class">Button</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
