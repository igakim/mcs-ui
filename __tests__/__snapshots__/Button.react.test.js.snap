// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should be disabled and have "mcs-button-primary" class 1`] = `
<button
  className="mcs-button mcs-button-primary"
  disabled={true}
  type=""
>
  Button
</button>
`;

exports[`component should be disabled and have "mcs-button-primary" class 2`] = `
<button
  className="mcs-button mcs-button-primary"
  disabled="disabled"
  type=""
>
  Button
</button>
`;

exports[`component should be disabled and have "mcs-button-primary" class 3`] = `
<button
  className="mcs-button mcs-button-primary"
  disabled={true}
  type=""
>
  Button
</button>
`;

exports[`component should have "mcs-button-primary" class 1`] = `
<button
  className="mcs-button mcs-button-primary"
  disabled={false}
  type=""
>
  Button
</button>
`;

exports[`component should have user classes as well 1`] = `
<button
  className="mcs-button mcs-button-primary super-class"
  disabled={false}
  type=""
>
  Button
</button>
`;

exports[`component should match snapshot 1`] = `
<button
  className="mcs-button"
  disabled={false}
  type=""
>
  Button
</button>
`;
